### ISSUE TITLE
Incorrect vendored version of jsoniter

### ISSUE THREAD

**Issue by wfernandes** [operator-framework/getting-started#16]
Upon following the instructions to build the docker image for the `memcached-operator` I get the following error.

```bash
$ operator-sdk build quay.io/example/memcached-operator:v0.0.1
Error: failed to build: (building memcached-operator...
# github.com/example-inc/memcached-operator/vendor/k8s.io/apimachinery/pkg/runtime/serializer/json
vendor/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go:104:16: unknown field 'CaseSensitive' in struct literal of type jsoniter.Config
# github.com/example-inc/memcached-operator/vendor/k8s.io/client-go/util/workqueue
vendor/k8s.io/client-go/util/workqueue/delaying_queue.go:48:25: clock.Tick undefined (type clock.Clock has no field or method Tick)
)
```

After taking a look at the `Gopkg.lock`, it contains an older version of `json-iterator`.
```
[[projects]]
  name = "github.com/json-iterator/go"
  packages = ["."]
  revision = "ca39e5af3ece67bbcda3d0f4f56a8e24d9f2dad4"
  version = "1.1.3"
```
Unfortunately, version `1.1.3` doesn't have the case sensitive stuff. I suggest upgrading it to the [latest version](https://github.com/json-iterator/go/commit/f2b4162afba35581b6d4a50d3b8f34e33c144682) which currently contains some fixes for case sensitivity.

**Comment by nikhita** [comment]
@wfernandes It is correct that the currently tagged version of json-iter does not contain the case sensitive fix but the version of client-go that's been pinned does not have the case sensitive fix as well:

https://github.com/operator-framework/operator-sdk/blob/280ebfaef7d43ff6712fe7358480c69e750a29cc/Gopkg.lock#L401-L402

Also, the fix has still not been cherry-picked to a 1.9 patch release: https://github.com/kubernetes/kubernetes/pull/65298.

I tried the example operator https://github.com/operator-framework/operator-sdk#quick-start and it worked for me. :thinking:

**Comment by nikhita** [comment]
@wfernandes can you try `make dep` before the operator-sdk commands?
